realm/realm-library/src/main/cpp/io_realm_SyncManager.cpp

```diff
@@ -96,3 +96,23 @@ JNIEXPORT void JNICALL Java_io_realm_SyncManager_nativeConfigureMetaDataSystem(J
     }
     CATCH_STD()
 }
+
+JNIEXPORT void JNICALL Java_io_realm_SyncManager_nativeSimulateSyncError(JNIEnv* env, jclass, jstring localRealmPath,
+                                                                         jint errorCode, jstring errorMessage,
+                                                                         jboolean isFatal)
+{
+    TR_ENTER()
+    try {
+        JStringAccessor local_realm_path(env, localRealmPath);
+        JStringAccessor error_message(env, errorMessage);
+
+        std::shared_ptr<SyncSession> session = SyncManager::shared().get_existing_active_session(local_realm_path);
```

Comment:
I know, but perhaps I just like actually having the type information, because then I don't have to think when reading it :) So some sort of guideline when to use `auto` vs. not would be nice?

Anyway, I'll change it.